src/query.js

```diff
@@ -18,7 +18,7 @@ const createQuery = query => (params, options) =>
 /**
  * @private
  */
-const createQueryFor = curry((query, table) => compose(createQuery(query), addTableName(table)));
+const createQueryFor = curry((query, table) => useWith(createQuery(query), [addTableName(table)]));
```

Comment:
Although not strictly needed, `useWith` should declare a transformer function for each argument its `fn` function expects to avoid extra confusion.

```suggestion
const createQueryFor = curry((query, table) => useWith(createQuery(query), [addTableName(table), identity]));
```

Reply:
Adding the `identity` will force the call the resulting function with `query(params, options)`, because `query(params)` returns a function that expect the `options` as an argument and that will be a breaking change.

Reply:
I see. You're right. The problem is that `useWith` returns a curried function, so by adding `[identity]`, `query(params)` returns yet another function instead of invoking itself. And yes, that'd be a breaking change.

Question is: how do we make this match the standalone `query` call (the one you get if you _don't_ use `forTable`)?

Reply:
I'll update to use the first option, seems easy to follow and will respect the original query function.